os/arch/arm/src/common/up_restoretask.c

```diff
@@ -83,7 +83,7 @@ void up_restoretask(struct tcb_s *tcb)
 #endif
 
 #ifdef CONFIG_ARCH_USE_MMU
-		if (tcb->app_id && tcb->pgtbl != mmu_l1_pgtable()) {
+		if (tcb->app_id && tcb->pgtbl && tcb->pgtbl != mmu_l1_pgtable()) {
```

Comment:
It looks unnecessary. Previous covers this new addition as well in this case..

Reply:
No Chang. 
Previously we only checked `tcb->pgtbl != mmu_l1_pgtable()`.
This condition will be true even if pgtbl is NULL. But in NULL case, we must not execute the if statement. 
So, I think both checks are necessary.